### ISSUE TITLE
package.json "main" doesn't export module

### ISSUE THREAD

**Issue by Tokimon** [FND/tiny-script-loader#2]
Hi.
Great module that I have been using a lot, but I recently came across an error unfortunately that prevents me from using this module.

I have a build that is split in two: one to target newer browsers using ES6+, and one to target IE11 (yeah we have to).

The problem is that other modules have the strategy to refer to two different versions of their script in the package.json:
- **module**: the ES6+ compliant version of the script
- **main**: the CJS (ES5) compliant script

However your module refers to the minified version of the script in **main**, which exports nothing, so I cannot use it in my build.

It is really too bad and you should reconsider your file strategy to something like this:
- **module**: loader.js - module compliant version (as it is right now)
- **main**: loader.cjs.js - CJS compliant version
- _no entry_: loader.min.js - this can be used to be included directly into a script tag if needed

In any case this is suggestion that you can do with what you want, but it is currently keeping me from using the module.

**Comment by FND** [comment]
Thanks for reporting this. I hadn't imagined that anyone but me would actually be using this, so I'd be curious how you found out about it.

The idea was that `loader.min.js` could be used without any module system or build step; it establishes `loadScripts` as a (gasp!) global variable. That's also why `loadScript` (singular) is a property of `loadScripts`, by the way. There didn't seem to be a reason to provide a non-minified version of this.

`loader.js` on the other hand is the module which could be imported in any ESM-compatible environment (directly in modern browsers and Node v12+ or via a bundler like Rollup or webpack). In that case, your build process should take care of minification, if desired.

Given that ESM is well-established by now, CommonJS didn't seem necessary to me - so far, anyway. Is there a reason you don't use the ESM version in your imports?

Update: On second thought, I might have ignored the ES5 vs. ES6 aspect above. However, the source code is actually ES5-compatible, `loader.min.js` is [generated](https://github.com/FND/tiny-script-loader/blob/81ba8e05b28820552f294bffc8b6a7c3f3645843/package.json#L19) from `loader.js` simply by removing `export default` (plus minification). So your IE11 bundle (I sympathize) could safely use the module version as well.

**Comment by Tokimon** [comment]
As I tried to explain, I have two builds running: One for modern browsers and one for older browsers like IE 11. The problem is that we generally ignore **node_modules** in our babel transpilation and as such your ESM (ESM = ES Module right?) code is not transformed into ES 5 for the older browsers and therefore it breaks the code once ran in those environments.

I have worked around the problem by pointing to the cjs file direcly, but it is less ideal. If you look at a package like [react-spring](https://github.com/react-spring/react-spring/blob/master/package.json) they have done what I propose and it seems like a reasonable solution. In that way I can just make sure that webpack looks at the correct entry for each of the builds.

Your minimized version is fine if you include the file directly into a html page, but it doesn't work in an module like environment. This is especially evident when I use one source for both builds, but they treat the code differently (export vs defining in global space), which makes it difficult to manage. so having the two different versions both export (module = module export, main = cjs export) makes the code easier to reason about 🙂

**Comment by Tokimon** [comment]
> so I'd be curious how you found out about it.

Actually I can't remember how I found you module, but I was searching for a small and simple script-loader and somehow i stumbled upon this one and it did exactly what i needed. So great job :+1:
You could register it on http://microjs.com 🙂

**Comment by FND** [comment]
> code is not transformed into ES 5 for the older browsers and therefore it breaks the code once ran in those environments

This is what I don't understand: This library's source code is written in ES5 - with the sole exception of `export default` at the beginning, which should be stripped by the bundler (webpack in your case; I use Rollup instead). Thus transpiling shouldn't be necessary here because the result would stay the same - what was the error in IE11?

(Also see the "On second thought" update I'd added to above's original response later on.)

PS: I'm not sure this library qualifies for microjs.com: https://github.com/microjs/microjs.com#rules - you might try though; anyone can submit a library.

**Comment by Tokimon** [comment]
Yes. Sorry you are right. I got confused there for a second. I apologize.

So the real problem here is the confusion between `module` and `main`. In my build I have to determine the entry depending on the type of build I have to accommodate the way fx react-spring does it. But the I run into the problem where your package has two different entries and `main` is not exporting anything which ends up colliding with the way I am including it.

So. I did a quick test and having the `main` entry point to `loader.js` solves the issue. Maybe you don't need the `module` entry?

**Comment by FND** [comment]
No need to apologize; I got a little confused there myself.

In fact, you've made realize that listing `loader.min.js` as `main` is probably wrong; that's what the [`browser` field](https://docs.npmjs.com/files/package.json#browser) is for, I think. So I'm inclined to go with this instead:

```json
    "main": "loader.js",
    "type": "module",
    "browser": "loader.min.js",
```

Would that work for you?

Again, thanks for your input!

**Comment by Tokimon** [comment]
Hey. Sorry for the delay. The notification got lost  in my list.
I guess that could work yes. 👍

**Comment by FND** [comment]
done: 3b1d4451bfbd179a5e49854d93fed6026ccba670 - released as v2.0.0
